components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3271,7 +3417,12 @@ subroutine mam_gasaerexch_1subarea(                           &
          iaer = iaer_nh4 ; igas = igas_nh3
          do n = 1, n_mode
             if (uptkaer(igas,n) <= 0.0_r8) cycle
-            tmpa = qaer_cur(iaer,n) - 2.0_r8*qaer_cur(iaer_so4,n)
+!++hybrown
+!            tmpa = qaer_cur(iaer,n) - 2.0_r8*qaer_cur(iaer_so4,n)
+            do ispec = 1, nso4
```

Comment:
This does not look quite right, you loop over ispec but tmpa looks like it just gets reset, so on loop exit it should equal qaer_cur(iaer,n) - 2.0_r8*qaer_cur(iaer_so4(nso4),n)